sql/core/src/main/scala/org/apache/spark/sql/expressions/udaf.scala

```diff
@@ -129,6 +129,13 @@ abstract class UserDefinedAggregateFunction extends Serializable {
         isDistinct = true)
     Column(aggregateExpression)
   }
+
+  /**
+    * The name of the UDAF. This is currently the simple name of class. This can create an
+    * [[java.lang.InternalError]] if the UDAF class was created in the REPL; override this method
+    * in these cases.
+    */
+  def name: String = getClass.getSimpleName
```

Comment:
I am wondering if we can catch the error and re-throw it with a more informative error message? In our error message, we can let the user know he/she needs to override `def name` to workaround that Scala issue.